ui/src/message_popup/renderers/read_more.jsx

```diff
@@ -0,0 +1,64 @@
+/* @flow weak */
+import React from 'react';
+
+/*
+  Component that displays a text block that can be expanded or collapsed.
+*/
+export default React.createClass({
```

Comment:
This is a nice way to factor out the little bit of transient state that is needed to track expanded/collapsed! 👍